eng/New-TestResources.ps1

```diff
@@ -108,7 +129,10 @@ if ($ProvisionerApplicationId) {
     Log "Logging into service principal '$ProvisionerApplicationId'"
     $provisionerSecret = ConvertTo-SecureString -String $ProvisionerApplicationSecret -AsPlainText -Force
     $provisionerCredential = [System.Management.Automation.PSCredential]::new($ProvisionerApplicationId, $provisionerSecret)
-    $provisionerAccount = Connect-AzAccount -Tenant $TenantId -Credential $provisionerCredential -ServicePrincipal
+
+    Retry {
+        $script:provisionerAccount = Connect-AzAccount -Tenant $TenantId -Credential $provisionerCredential -ServicePrincipal

```

Comment:
Since `Retry` is returning the result of the script block's invocation, could this also be phrased as: 

```
$provisionerAccount = Retry { Connect-AzAccount -Tenant $TenantId -Credential $provisionerCredential -ServicePrincipal } 
```

Reply:
True, though I originally hadn't planned it that way. Still, there's nothing wrong with this syntax. IF there's no other changes, I wouldn't see a point in changing just this (or even if there were).